index.js

```diff
@@ -1050,131 +1136,209 @@
-      case 'Array#fantasy-land/concat':
-        return Array$prototype$concat;
-      case 'Array#fantasy-land/filter':
-        return Array$prototype$filter;
-      case 'Array#fantasy-land/map':
-        return Array$prototype$map;
-      case 'Array#fantasy-land/ap':
-        return Array$prototype$ap;
-      case 'Array#fantasy-land/chain':
-        return Array$prototype$chain;
-      case 'Array#fantasy-land/alt':
-        return Array$prototype$alt;
-      case 'Array#fantasy-land/reduce':
-        return Array$prototype$reduce;
-      case 'Array#fantasy-land/traverse':
-        return Array$prototype$traverse;
-      case 'Array#fantasy-land/extend':
-        return Array$prototype$extend;
-      case 'Arguments#fantasy-land/equals':
-        return Arguments$prototype$equals;
-      case 'Arguments#fantasy-land/lte':
-        return Arguments$prototype$lte;
-      case 'Error#fantasy-land/equals':
-        return Error$prototype$equals;
-      case 'Object#fantasy-land/equals':
-        return (Object.values (value)).every (Setoid.test) ?
-          Object$prototype$equals :
-          null;
-      case 'Object#fantasy-land/lte':
-        return (Object.values (value)).every (Ord.test) ?
-          Object$prototype$lte :
-          null;
-      case 'Object#fantasy-land/concat':
-        return Object$prototype$concat;
-      case 'Object#fantasy-land/filter':
-        return Object$prototype$filter;
-      case 'Object#fantasy-land/map':
-        return Object$prototype$map;
-      case 'Object#fantasy-land/ap':
-        return Object$prototype$ap;
-      case 'Object#fantasy-land/alt':
-        return Object$prototype$alt;
-      case 'Object#fantasy-land/reduce':
-        return Object$prototype$reduce;
-      case 'Object#fantasy-land/traverse':
-        return Object$prototype$traverse;
-      case 'Function#fantasy-land/equals':
-        return Function$prototype$equals;
-      case 'Function#fantasy-land/compose':
-        return Function$prototype$compose;
-      case 'Function#fantasy-land/map':
-        return Function$prototype$map;
-      case 'Function#fantasy-land/promap':
-        return Function$prototype$promap;
-      case 'Function#fantasy-land/ap':
-        return Function$prototype$ap;
-      case 'Function#fantasy-land/chain':
-        return Function$prototype$chain;
-      case 'Function#fantasy-land/extend':
-        return Function$prototype$extend;
-      case 'Function#fantasy-land/contramap':
-        return Function$prototype$contramap;
-      default:
-        return null;
+    // Single-member types are identified most quickly.
+    switch (value) {
+      case null: return prototypeImplementations.Null[name];
+      case undefined: return prototypeImplementations.Undefined[name];
+    }
+
+    // Check if we can dispatch to a Fantasy Land method.
+    var prefixedName = 'fantasy-land/' + name;
+    var condition =
+      value.constructor == null || value.constructor.prototype !== value;
```

Comment:
I remember this being important, but I couldn't remember why. I called it `condition` so we can discuss a better name that might say something about why this check is needed.

Reply:
#22